mod/ckeditor/views/default/ckeditor/init.php

```diff
@@ -7,12 +7,14 @@
 
 ?>
 <script>
-// This global variable must be set before the editor script loading.
-CKEDITOR_BASEPATH = elgg.config.wwwroot + 'mod/ckeditor/vendors/ckeditor/';
+require(['elgg'], function (elgg) {
+	// This global variable must be set before the editor script loading.
```

Comment:
Plugins adding values to elgg.config server-side would simplify some things, but also invite devs to implicitly depend on a global, synchronous data source. I wouldn't be against adding an `elgg/plugin_settings` AMD module that plugins could pull data out of. We could still define this in js/initialize_elgg for performance.